### PR TITLE
chore: support experimentalSingleTabRunMode in protocol

### PR DESCRIPTION
**Pull request by mschile**
### Additional details

* When `experimentalSingleTabRunMode` is enabled, the same tab is used for multiple specs. As such, the `protocolManager.connectToBrowser` method is not called after the first spec causing all CDP events to be ignored. In order to resolve the issue, we call `protocolManager.connectToBrowser` before the subsequent specs are run.
* Moved the protocol system-tests to their own project and added CT tests.

### Steps to test

Run multiple CT specs and verify the CDP events are captured.

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/server/lib/modes/run.ts [hunk 1]
```diff
@@ -492,6 +492,12 @@ async function waitForBrowserToConnect (options: { project: Project, socketId: s
       openProject.updateTelemetryContext(JSON.stringify(telemetry.getActiveContextObject()))
     }
 
+    // since we aren't going to be opening a new tab,
+    // we need to tell the protocol manager to reconnect to the existing browser
+    if (options.browser.family === 'chromium') {
```
Comment (ryanthemanuel) [inline packages/server/lib/modes/run.ts]: Let's do this if the protocolManager is there rather than relying on the browser family.

**Review by ryanthemanuel — APPROVED** [review]
Approved with the above suggestion

File: packages/server/lib/modes/run.ts (hunk 1 above)
Comment (mschile) [reply packages/server/lib/modes/run.ts]: Updated: [d586c22](https://github.com/cypress-io/cypress/pull/27659/commits/d586c225ca0c4e4d2a4344d43d2cddd0b1ffb091)

File: packages/server/test/unit/browsers/browsers_spec.js [hunk 2]
```diff
@@ -397,6 +397,12 @@ describe('lib/browsers/index', () => {
       })
     })
   })
+
+  context('connectProtocolToBrowser', () => {
+    it('calls connectProtocolToBrowser on the launcher', () => {
+      sinon.s
```
Comment (AtofStryker) [inline packages/server/test/unit/browsers/browsers_spec.js]: is there something missing here?

File: system-tests/projects/protocol/cypress/support/commands.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// ***********************************************
```
Comment (AtofStryker) [inline system-tests/projects/protocol/cypress/support/commands.ts]: can we just do `supportFile: false` in the configuration for this and remove this file?

File: system-tests/test/protocol_spec.js [hunk 4]
```diff
@@ -50,7 +49,31 @@ describe('capture-protocol', () => {
...
-        systemTests.snapshot('protocol events', normalizeEvents(protocolEvents))
+        systemTests.snapshot('e2e events', normalizeEvents(protocolEvents))
+
+        fs.removeSync(getFilePath('e9e81b5e-cc58-4026-b2ff-8ae3161435a6.db'))
+      })
+    })
+  })
+
+  describe('component', () => {
+    it('verifies the protocol events are correct', function () {
```
Comment (AtofStryker) [inline system-tests/test/protocol_spec.js]: should we have two tests here, one for `experimentalSingleTabRunMode` on and one with it off?

File: system-tests/projects/protocol/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (AtofStryker) [inline system-tests/projects/protocol/tsconfig.json]: more of a nit: we should be able to target at least es6 here if not later

File: system-tests/projects/protocol/package.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "@packages/app-capture-protocol",
```
Comment (AtofStryker) [inline system-tests/projects/protocol/package.json]: something like this maybe?
```suggestion
  "name": "protocol-sample-project",
```

File: packages/server/test/unit/browsers/browsers_spec.js (hunk 2 above)
Comment (mschile) [reply packages/server/test/unit/browsers/browsers_spec.js]: Removed: [73fef85](https://github.com/cypress-io/cypress/pull/27659/commits/73fef8513d89e7d92f15a74d51467f8ad42e086c)

File: system-tests/projects/protocol/cypress/support/commands.ts (hunk 3 above)
Comment (mschile) [reply system-tests/projects/protocol/cypress/support/commands.ts]: Updated to `supportFile: false`, [73fef85](https://github.com/cypress-io/cypress/pull/27659/commits/73fef8513d89e7d92f15a74d51467f8ad42e086c)

File: system-tests/test/protocol_spec.js (hunk 4 above)
Comment (mschile) [reply system-tests/test/protocol_spec.js]: Good idea, updated to also test with the experiment off. [73fef85](https://github.com/cypress-io/cypress/pull/27659/commits/73fef8513d89e7d92f15a74d51467f8ad42e086c)

File: system-tests/projects/protocol/tsconfig.json (hunk 5 above)
Comment (mschile) [reply system-tests/projects/protocol/tsconfig.json]: Updated to `es6`: [73fef85](https://github.com/cypress-io/cypress/pull/27659/commits/73fef8513d89e7d92f15a74d51467f8ad42e086c)

File: system-tests/projects/protocol/package.json (hunk 6 above)
Comment (mschile) [reply system-tests/projects/protocol/package.json]: Updated to `protocol-sample-project`: [73fef85](https://github.com/cypress-io/cypress/pull/27659/commits/73fef8513d89e7d92f15a74d51467f8ad42e086c)

**Review by AtofStryker — APPROVED** [review]
looks good besides the system-test failures. We should be g2g once those are fixed

**Pull request merged**